File: Aditya/demandSimulate.m

function priceMatrix = demandSimulate(alpha, K0, sigma, nstep, nsim, maturity, P0)
    
    dt = maturity/nstep;
    priceMatrix = zeros(nsim,nstep+1);
    priceMatrix(:,1)= P0;

    dW = normrnd(0,1, nsim, nstep)*sqrt(dt);
    for iStep=2:nstep+1
        priceMatrix(:,iStep) = priceMatrix(:,iStep-1) + alpha*(K0 - priceMatrix(:,iStep-1))*dt + sigma*dW(:,iStep-1);
    end
    
end

File: Aditya(Python)/demandSimulate.py
import numpy as np
# demandSimulate.m
def demandSimulate(α, K0, σ, n_step, n_sim, maturity, P0):
    
    dt = maturity/n_step
    priceMatrix = np.zeros((n_sim, n_step+1))
    priceMatrix[:,0] = np.ones(n_sim) * P0
    
    dW = np.random.normal(0,1,size = (n_sim,n_step) ) * np.sqrt(dt)
    
    for i in range(1,n_step+1):
        priceMatrix[:,i] = priceMatrix[:,i-1] + α * (K0 - priceMatrix[:,i-1]) * dt + σ * dW[:,i-1]
        
    return priceMatrix